server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3879,6 +3879,8 @@ private void fillVMOrTemplateDetailOptions(final Map<String, List<String>> optio
 
         if (HypervisorType.KVM.equals(hypervisorType)) {
             options.put(VmDetailConstants.ROOT_DISK_CONTROLLER, Arrays.asList("osdefault", "ide", "scsi", "virtio"));
+            options.put(VmDetailConstants.VIDEO_HARDWARE, Arrays.asList("cirrus", "vga", "qxl", "virtio"));
+            options.put(VmDetailConstants.VIDEO_RAM, Arrays.asList("16384", "32768", "65536"));
```

Comment:
@leolleeooleo 
can videoram be any integer value ?
if so, you can use
```
options.put(VmDetailConstants.VIDEO_RAM, Collections.emptyList());
```

Reply:
@leolleeooleo 
I was not asking you to change the values. sorry for confusion. 
I just wanted to know if there is any restriction here . for example, the min value, the max value, or it must be power of 2.

Reply:
Note that if VIDEO_RAM = 0, VIDEO_HARDWARE will not work.

If not power of 2, libvirt will auto make it to power of 2.

https://github.com/apache/cloudstack/blob/07f30f6867da2c98e44a61eaf2cfd21c2647725b/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java#L1631

Reply:
would it better to change to the following ?
```
            if (_videoModel != null && !_videoModel.isEmpty()) {
                videoBuilder.append("<video>\n");
                if (_videoRam != 0) {
                    videoBuilder.append("<model type='" + _videoModel + "' vram='" + _videoRam + "'/>\n");
                } else {
                    videoBuilder.append("<model type='" + _videoModel + "'/>\n");
                }
                videoBuilder.append("</video>\n");
                return videoBuilder.toString();
            }
```

Reply:
It is a good idea.
I was worry about it if videoRam is not provided.
I had test it with no error, and here are the document.
https://libvirt.org/formatdomain.html#video-devices
If no value is provided the default is used.